// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option objc_class_prefix = "RTG";

package audiostream;

// Interface exported by the server.
service AudioStream {
  // A server-to client streaming RPC.
  //
  // Streams an audio file from the location specified by client.
  //
  // No audio is played if file is not found.
  rpc StreamAudio(FilePath) returns (stream AudioData) {}
}

// Path to the audio file is represented as a string.
message FilePath {
  string file_path = 1;
}

// A latitude-longitude rectangle, represented as two diagonally opposite
// points "lo" and "hi".
message AudioData {
  // Array of audio data
  bytes audio_data = 1;
}
